File: IT/Containers/Esercizi/Java/Esercizio6.java
/* file Esercizio6.java */

import java.util.ArrayList;
import java.util.Collections;

public class Esercizio6 {
	public static void main(String [] args) {
		ArrayList<Integer> nums = new ArrayList<Integer>();

		nums.add(1);
		nums.add(3);
		nums.add(5);
		nums.add(7);
		nums.add(9);

		System.out.println(nums);

		Collections.reverse(nums);

		System.out.println(nums);
	}
}

File: IT/Containers/Esercizi/C++/esercizio6.cpp
/*
file esercizio6.cpp
compilare con: g++ -std=c++11 esercizio6.cpp
*/

#include <iostream>
#include <vector>
#include <iterator>
#include <algorithm>

using std::cout;
using std::endl;
using std::vector;
using std::ostream_iterator;
using std::reverse;

int main()
{
	vector<int> nums = {1, 3, 5, 7, 9};

	cout << "[";
	copy(nums.begin(), nums.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	reverse(nums.begin(), nums.end());

	cout << "[";
	copy(nums.begin(), nums.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;
}

